docs/core/fees.md

```diff
@@ -0,0 +1,452 @@
+> the global state (unless manually topped up).
+
+In the process of creating a new account, you must ensure you deposit enough
+lamports to be above this minimum balance. Anything lower that this minimum
+threshold will result in a failed transaction.
+
+Every time an account's balance is reduced, the runtime performs a check to see
+if the account will still be above this minimum balance for rent exemption.
+Unless they reduce the final balance to exactly `0` (closing the account),
+transactions that would cause an account's balance to drop below the rent exempt
+threshold will fail.
+
+Since the specific minimum balance for an account to become rent exempt is
+dependant on the blockchain's current [rent rate](#rent-rate) and the desired
+amount of storage space an account wants to allocate (account size), it is
+recommended to use the
+[`getMinimumBalanceForRentExemption`](/docs/rpc/http/getMinimumBalanceForRentExemption.mdx)
+RPC endpoint to calculate the specific balance for a given account size.
+
+The required rent deposit amount can also be estimated via the
+[`solana rent` CLI subcommand](https://docs.solanalabs.com/cli/usage#solana-rent):
+
+```shell
+solana rent 15000
+
+# output
+Rent per byte-year: 0.00000348 SOL
+Rent per epoch: 0.000288276 SOL
+Rent-exempt minimum: 0.10529088 SOL
+```
+
+### Garbage collection
+
+Accounts that do not maintain a lamport balance greater than zero are removed
+from the network in a process known as _garbage collection_. This process is
+done to help reduce the network wide storage of no longer used/maintained data.
+
+After a transaction successfully reduces an accounts balance to exactly `0`,
+garbage collection happens automatically by the runtime. Any transaction that
+attempts to reduce an accounts balance lower that its minimum balance for rent
+exemption (that is not exactly zero) will fail.
+
+> Even after an account has been removed from the network (via garbage
+> collection), it may still have transactions associated with it's address
+> (either past history or in the future). Even though a Solana block explorer
+> may display an "account not found" type of message, you may still be able to
+> view transaction history associated with that account.
+
+You can read the validator
+[implemented proposal](https://docs.solanalabs.com/implemented-proposals/persistent-account-storage#garbage-collection)
+for garbage collection to learn more.
+
+## Compute Budget
+
+To prevent abuse of computational resources, each transaction is allocated a
+compute budget. The budget specifies a maximum number of compute units that a
+transaction can consume, the costs associated with different types of operations
+the transaction may perform, and operational bounds the transaction must adhere
+to.
+
+As the transaction is processed compute units are consumed by its instruction's
+programs performing operations such as executing SBF instructions, calling
```

Comment:
This could just be 'executing instructions'. as we haven't introduced the term 'SBF' yet and it's not in terminology as we want to get rid of it soon.

Reply:
Good callout. I have also not gotten to updating this section yet: "WIP draft"